src/python/pants/backend/codegen/protobuf/python/python_protobuf_subsystem.py

```diff
@@ -59,9 +61,42 @@ def runtime_dependencies(self) -> UnparsedAddressInputs:
     def mypy_plugin(self) -> bool:
         return cast(bool, self.options.mypy_plugin)
 
-    @property
-    def mypy_plugin_version(self) -> str:
-        return cast(str, self.options.mypy_plugin_version)
+
+class PythonProtobufMypyPlugin(PythonToolRequirementsBase):
+    options_scope = PythonProtobufSubsystem.subscope("mypy-plugin")
```

Comment:
FYI this is the first usage of subscopes in Pants 2.x. I think we had some tentative discussion a few months back on whether we want to have this or not, and what the semantics should be. Cc @stuhood, you had some thoughts iirc.

I think this is fine, but want to draw attention to this setting up a new precedent.

Reply:
I think that this usecase is roughly in the spirit of how we would want subscopes to be used. But honestly, I'm mostly surprised that it works.

The primary difference to watch out for is that in the past, `Subscopes` were primarily generated dynamically to add a common/reusable `Subsystem` to a scope. Because of that, we have inheritance/recursion of options applied to scopes. For example: with a scope `my-jvm-tool.common-jvm-config`, setting an option in scope `my-jvm-tool.common-jvm-config` affects only that instance of the subsystem. But setting that same option in `common-jvm-config` would affect _all_ subscopes like `*.common-jvm-config`.

Reply:
Yeah - I wrote that bit of Subsystems IIRC. This does not use that though, it just uses the subscope namespace as a trick to ensure shadowing as opposed to setting the subsystem options scope to 'python-protobuf-mypy-plugin' which I also could have done. The difference would have merely been `s/./-/` in the pants.toml config key name. The `s/./-/` does not affect either the flag name or the env var name.